marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -367,11 +372,21 @@ marlowePlutusContract = selectList [create, apply, auto, redeem, close]
                     Transition _ _ marloweData -> continueWith marloweData
                     InitialState _ marloweData -> continueWith marloweData
 
-            CloseContract -> do
+            CloseContract payments -> do
                 logInfo @String $ "CloseContract"
-                tell $ OK reqId "auto"
-                marlowePlutusContract
+                let closeContract = do
+                        marloweData <- SM.runStep theClient (slotRange, [])
+                        case marloweData of
+                            SM.TransitionFailure e -> throwing _TransitionError e
+                            SM.TransitionSuccess d -> do
+                                tell $ OK reqId "auto"
+                                marlowePlutusContract
 
+                catching _StateMachineError closeContract $ \err -> do
```

Comment:
What type of error should make us try again in two slots? Shouldn't we explicitly match against a particular error?